index.js

```diff
@@ -35,10 +35,26 @@ function createVNode(domNode, key) {
 }
 
 createVNode.fromHTML = function(html, key) {
-  var domNode = document.createElement('div'); // create container
-  domNode.innerHTML = html; // browser parses HTML into DOM tree
-  var child = domNode.children.length ? domNode.children[0] : domNode.firstChild;
-  return createVNode(child, key);
+  var rootNode = null;
+
+  try {
+    // Everything except iOS 7 Safari, IE 8/9, Andriod Browser 4.1/4.3
+    var parser = new DOMParser();
+    var doc = parser.parseFromString(html, 'text/html');
+    rootNode = doc.documentElement;
+  } catch(e) {
+    // Old browsers
+    var ifr = document.createElement('iframe');
+    ifr.setAttribute('data-content', html);
+    ifr.src = 'javascript: window.frameElement.getAttribute("data-content");';
```

Comment:
@marcelklehr The HTML is inserted using a little trick to get old browsers working. The URL of the page in the `<iframe>` is set to `javascript: window.frameElement.getAttribute("data-content");`. This tells the browser to set the page's source to be equal to the frameElement's attribute `data-content`, which contains the HTML to be parsed.